pkg/registration/hub/manifests/rbac/managedcluster-watch-clusterrole.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:managedcluster:{{ .ManagedClusterName }}-watch
+  labels:
```

Comment:
can we just update the existing clusterrole? So we do not recreate another two additional clusterroles

Reply:
Yes, that's also an approach, the PR updated, some original concerns are:
1. the following four rbac files used to have the same lifecycle, change the `clusterrole.yaml` need to break this set into 2 sets, the different is clusterrole doe
```go
var ClusterSpecificRBACFiles = []string{
	"rbac/managedcluster-clusterrole.yaml",
	"rbac/managedcluster-clusterrolebinding.yaml",
	"rbac/managedcluster-registration-rolebinding.yaml",
	"rbac/managedcluster-work-rolebinding.yaml",
}
```

2. the `update` need to introduce another varible in process of `ManagedClusterAssetFn`, and then also affect the upper caller functions:

```go
func ManagedClusterAssetFn(fs embed.FS, managedClusterName string) resourceapply.AssetFunc {
	return func(name string) ([]byte, error) {
		config := struct {
			ManagedClusterName string
		}{
			ManagedClusterName: managedClusterName,
		}

		template, err := fs.ReadFile(name)
		if err != nil {
			return nil, err
		}
		return assets.MustCreateAssetFromTemplate(name, template, config).Data, nil
	}
}
```

Reply:
PR updated, if the new changes are accepted, I will squash the commits and continue on.